OpenRA.Mods.Common/Traits/Render/WithUnitBody.cs

```diff
@@ -0,0 +1,26 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public class WithUnitBodyInfo : WithSpriteBodyInfo, Requires<IFacingInfo>
+	{
+		public override object Create(ActorInitializer init) { return new WithUnitBody(init, this); }
+	}
+
+	public class WithUnitBody : WithSpriteBody
+	{
+		public WithUnitBody(ActorInitializer init, WithUnitBodyInfo info)
+			: base(init, info) { }
```

Comment:
With my earlier comment I meant passing the facing function from here.  Otherwise, the `Requires<IFacingInfo>` above doesn't make sense, because this trait doesn't (explicitly) use IFacing.